poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java

```diff
@@ -1958,80 +1943,13 @@ void initDrawings() {
      * @see #PICTURE_TYPE_JPEG
      * @see #PICTURE_TYPE_DIB
      */
-    @SuppressWarnings("fallthrough")
     @Override
     public int addPicture(byte[] pictureData, int format) {
-        initDrawings();
-
-        byte[] uid = DigestUtils.md5(pictureData);
-        EscherBlipRecord blipRecord;
-        int blipSize;
-        short escherTag;
-        switch (format) {
-            case PICTURE_TYPE_WMF:
-                // remove first 22 bytes if file starts with the WMF placeable header
-                if (FileMagic.valueOf(pictureData) == FileMagic.WMF) {
-                    pictureData = IOUtils.safelyClone(pictureData, 22, pictureData.length - 22, MAX_IMAGE_LENGTH);
-                }
-                // fall through
-            case PICTURE_TYPE_EMF:
-                EscherMetafileBlip blipRecordMeta = new EscherMetafileBlip();
-                blipRecord = blipRecordMeta;
-                blipRecordMeta.setUID(uid);
-                blipRecordMeta.setPictureData(pictureData);
-                // taken from libre office export, it won't open, if this is left to 0
-                blipRecordMeta.setFilter((byte) -2);
-                blipSize = blipRecordMeta.getCompressedSize() + 58;
-                escherTag = 0;
-                break;
-            default:
-                EscherBitmapBlip blipRecordBitmap = new EscherBitmapBlip();
-                blipRecord = blipRecordBitmap;
-                blipRecordBitmap.setUID(uid);
-                blipRecordBitmap.setMarker((byte) 0xFF);
-                blipRecordBitmap.setPictureData(pictureData);
-                blipSize = pictureData.length + 25;
-                escherTag = (short) 0xFF;
-                break;
```

Comment:
Am I missing something about any of this being defaulted? Or is it altogether unnecessary?

Reply:
Combination of defaulted elsewhere, previously redundant assignments, and no longer necessary, _I think_. We'll want to verify xls w/ images thoroughly in the product